packages/adapter-utils/src/utils.ts

```diff
@@ -705,3 +705,27 @@ export const getAllReferencedIds = (element: Element): Set<string> => {
 
   return allReferencedIds
 }
+
+export const getNestedStaticFiles = (value: Value): StaticFile[] => {
+  if (isElement(value)) {
+    const allReferencedIds = new Set<StaticFile>()
+    const transformFunc: TransformFunc = ({ value: val }) => {
+      if (isStaticFile(val)) {
+        allReferencedIds.add(val)
+      }
+      return val
+    }
+    transformElement({ element: value, transformFunc, strict: false })
+    return wu(allReferencedIds.values()).toArray()
+  }
+  if (_.isArray(value)) {
+    return _.flatten(value.map(getNestedStaticFiles))
```

Comment:
nit (also for the statement below): 
```suggestion
    return value.flatMap(getNestedStaticFiles)
```

Reply:
cool